Formula/vlang.rb

```diff
@@ -0,0 +1,14 @@
+class Vlang < Formula
+  desc "V programming language"
+  homepage "https://vlang.io"
+  url "https://github.com/vlang/v/archive/0.1.18.tar.gz"
+  sha256 "3f3407a78aca7fc3b42a3fc1f1d2b9724c1e4c71fbd5d37ff12976cd2305cec1"
+
+  def install
+    system "make", "sudo v symlink"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/v -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
V is a new programming language compiler. It has tests, and you can run them by (after make has run):
`./v test v`
I am not familiar with the brew expectations/ formula syntax so can not comment how that should be done in brew.